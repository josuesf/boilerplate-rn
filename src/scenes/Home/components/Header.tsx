import * as React from 'react';
import {Box, Text} from '@components';
import {useTranslation} from 'react-i18next';
interface Props {
  nameUser: string;
}
export const Header = ({nameUser}: Props) => {
  const {t} = useTranslation();
  return (
    <Box mb="sm">
      <Text variant="title">{t('home.header.title')}</Text>
      <Text testID="name-user">{nameUser}</Text>
    </Box>
  );
};
