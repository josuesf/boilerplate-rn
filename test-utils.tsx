import React from 'react';
import {render as rntlRender} from '@testing-library/react-native';
import {ThemeProvider} from './@components';
import moment from 'moment';
import 'moment/locale/es-mx';
import {Provider} from 'react-redux';
import {store} from 'src/redux/store';

function render(ui: React.ReactElement) {
  moment.locale('es-mx');
  return rntlRender(ui, {
    wrapper: ({children}) => (
      <ThemeProvider>
        <Provider store={store}>{children}</Provider>
      </ThemeProvider>
    ),
  });
}

export * from '@testing-library/react-native';

export {render};
