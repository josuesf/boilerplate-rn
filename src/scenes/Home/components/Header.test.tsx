import * as React from 'react';
import {Header} from './Header';
import {render, screen} from 'test-utils';

it('renders Header correctly', async () => {
  const nameExpected = 'Josue Farfan';
  const {queryByTestId} = render(<Header nameUser={nameExpected} />);
  expect(queryByTestId('name-user')).toHaveTextContent(nameExpected);
  expect(screen.toJSON()).toMatchSnapshot();
});
