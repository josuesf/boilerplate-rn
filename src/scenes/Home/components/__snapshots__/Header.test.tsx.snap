// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 39,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#020202",
          "fontFamily": "Avenir",
          "fontSize": 39,
          "fontWeight": "800",
          "lineHeight": 53,
        },
      ]
    }
  >
    home.header.title
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#020202",
          "fontFamily": "Avenir",
          "fontSize": 31,
          "lineHeight": 39,
        },
      ]
    }
    testID="name-user"
  >
    Josue Farfan
  </Text>
</View>
`;
